src/vivarium_csu_alzheimers/components/population.py

```diff
@@ -0,0 +1,157 @@
+    """This is using FertilityCrudeBirthRate from Vivarium Public Health as a template."""
+
+    #####################
+    # Lifecycle methods #
+    #####################
+
+    def setup(self, builder: Builder) -> None:
+        self.clock = builder.time.clock()
+        self.randomness = builder.randomness  # Manager
+        # NOTE: All three of these methods are capping the upper age bound at 100
+        self.incidence_rate = self.load_incidence_rate(builder)
+        self.pop_structure = self.load_population_structure(builder)
+        prevalence = self.load_prevalence(builder)
+        # Model scale = (population_size / (pop_structure * prevalence).sum())
+        self.model_scale = builder.configuration.population.population_size / (
+            (self.pop_structure * prevalence).sum()
+        )
+        self.simulant_creator = builder.population.get_simulant_creator()
+        self.age_start = builder.configuration.population.initialization_age_min
+        self.age_end = builder.configuration.population.initialization_age_max
+
+    ########################
+    # Event-driven methods #
+    ########################
+
+    def on_time_step(self, event: Event) -> None:
+        """Adds new simulants every time step based on the Crude Birth Rate
+        and an assumption that birth is a Poisson process
+
+        Parameters
+        ----------
+        event
+            The event that triggered the function call.
+        """
+
+        step_size = utilities.to_years(event.step_size)
+        # TODO: get incidence rates and population for year in forecasted data if necessary
+        pop_structure = self.pop_structure.copy()
+        pop_structure.index = pop_structure.index.droplevel(["year_start", "year_end"])
+        mean_incident_cases = (
+            self.incidence_rate * pop_structure * step_size * self.model_scale
+        )
+        simulants_to_add = pd.Series(0, index=mean_incident_cases.index)
+
+        # Determine number of simulants to add for each demographic group
+        for idx, value in mean_incident_cases.items():
+            sex, age_lower, age_upper = idx
+            r = np.random.RandomState(
+                seed=self.randomness.get_seed(f"f{age_lower}_{sex}_alz_incidence")
+            )
+            simulants_to_add[(sex, age_lower, age_upper)] = r.poisson(value)
+        total_simulants_to_add = simulants_to_add.sum()
+
+        if total_simulants_to_add > 0:
+            self.simulant_creator(
+                total_simulants_to_add,
+                {
+                    "sim_state": "time_step",
+                    "demographic_counts": simulants_to_add,
+                },
+            )
+
+    ##################
+    # Helper methods #
+    ##################
+
+    def load_incidence_rate(self, builder: Builder) -> pd.Series:
+        incidence_rate = builder.data.load(data_keys.ALZHEIMERS.INCIDENCE_RATE)
+        incidence_rate.loc[incidence_rate["age_end"] == 125, "age_end"] = 100
```

Comment:
Nit: 
```suggestion
        incidence_rate.loc[incidence_rate["age_end"] == 125, "age_end"] = self.age_end
```
Do the same in the other two places. You will need to set `self.age_end` earlier in `setup()`